### PR TITLE
Feat/button

### PR DESCRIPTION
**Pull request by Arthurmqz**
# Description

This update addresses the need for a customizable and reusable set of button components in the project.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Other:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/components/button/src/Button.tsx [hunk 1]
```diff
@@ -1,28 +1,39 @@
...
+  secondary: SecondaryButton,
+  tertiary: TertiaryButton,
+  error: ErrorButton,
+};
+
 const Button = ({
-  primary = false,
-  size = 'small',
-  backgroundColor,
+  typeBtn,
```
Comment (joaoprcosta) [inline packages/components/button/src/Button.tsx:18]: Type is more accurate and simple.

```suggestion
  type,
```

File: packages/components/button/src/__tests__/Button.spec.tsx [hunk 2]
```diff
@@ -1,15 +1,192 @@
...
+};
+
+describe('<Button >', () => {
+  it('should render a clickable primary button', () => {
+    renderTheBtnType('Primary button and available', 'primary', false);
+
+    const buttonElement = screen.getByRole('button', {
+      name: /Primary button and available/i,
+    });
+    expect(buttonElement).not.toBeDisabled();
```
Comment (joaoprcosta) [inline packages/components/button/src/__tests__/Button.spec.tsx:33]: The expect doesn't match with the message of the text. If you are just testing if the component is rendered, do not use a random prop like disable just to expect something.

File: packages/components/button/src/__tests__/Button.spec.tsx [hunk 3]
```diff
@@ -1,15 +1,192 @@
...
 import { ThemeProvider } from '@douro-ui/react';
+import { ButtonProps } from '../button.types';
 
-describe('<Button />', () => {
-  it('should render properly', () => {
-    render(
-      <ThemeProvider>
-        <Button label="Foo" />
-      </ThemeProvider>,
+const renderTheBtnType = (
```
Comment (joaoprcosta) [inline packages/components/button/src/__tests__/Button.spec.tsx:6]: Use the render method from test-utils. It already wraps the ThemeProvider.

Comment (joaoprcosta) [inline packages/components/button/src/__tests__/Button.spec.tsx:6]: You should create a reusable custom render to wrap all the providers (e.g. ThemeProvider in this case) and pass the components.

https://testing-library.com/docs/react-testing-library/setup/#custom-render

File: packages/components/button/src/__tests__/Button.spec.tsx [hunk 4]
```diff
@@ -1,15 +1,192 @@
...
+  });
+  it('should render a MD clickable tertiary button', () => {
+    renderTheBtnType('Tertiary button and available', 'tertiary', false, 'md');
+
+    const buttonElement = screen.getByRole('button', {
+      name: /Tertiary button and available/i,
+    });
+    expect(buttonElement).not.toBeDisabled();
+  });
+  it('should render a disabled tertiary button', () => {
```
Comment (joaoprcosta) [inline packages/components/button/src/__tests__/Button.spec.tsx:128]: I think you are repeating some tests, there is no real difference between a primary or secondary when testing a click. The behavior is the same depending on the type.

File: packages/components/button/src/Button.tsx (hunk 1 above)
Comment (Arthurmqz) [reply packages/components/button/src/Button.tsx:18]: If changing it to "type" causes a TypeScript error, we should consider an alternative name. Do you have any suggestions? Or can we keep "typeBtn"?

File: packages/components/button/src/buttonTypes/ErrorButton.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    <ButtonStyled
+      typeBtn={'error'}
+      size={size}
+      styled={mergedThemeValues}
+      onClick={onClick}
+      disabled={disabled}
+      data-testid={`button-${typeBtn}`}
+      {...props}
+    >
+      {label}
```
[the remark this answers is not in the document]
Comment (Arthurmqz) [reply packages/components/button/src/buttonTypes/ErrorButton.tsx:41]: Make sense, I'll replace it.

File: packages/components/button/src/__tests__/Button.spec.tsx (hunk 3 above)
Comment (Arthurmqz) [reply packages/components/button/src/__tests__/Button.spec.tsx:6]: I reviewed the comment, but it seems you might have looked at an older commit. The current commit is already wrapped and uses the correct props.

**Review by joaoprcosta — APPROVED** [review]

**Review by JuniorDiasOliveira — APPROVED** [review]

**Pull request merged**